### PR TITLE
Login form redesign

### PR DESCRIPTION
**Pull request by shsimeonova**

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.module.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+.loginFormHeader {
+  display: flex;
+  flex-direction: column;
+  text-align: center;
+}
+
+.loginFormLink {
+  color: $background-color;
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.module.scss:24]: suggestion: this color variable should be changed to primary-blue

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-auth.tsx [hunk 2]
```diff
@@ -85,6 +87,7 @@ const AuthProvider = ({ user, children }: IAuthProviderProps) => {
 
     const value = {
         user: internalUser,
+        loginErrorMessage,
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-auth.tsx:90]: As discussed before, this should be refactored to `state` & `actions`  when we do changes to this value.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/identity-config.ts [hunk 3]
```diff
@@ -3,13 +3,15 @@ import { WebStorageStateStore } from 'oidc-client';
...
 const authorityUrl = `${process.env.REACT_APP_PLATFORM_IDENTITY_BASE_URL}${process.env.REACT_APP_PLATFORM_IDENTITY_AUTHORITY_URL_POSTFIX}`;
+const authorityRegisterUrl = `${authorityUrl}/register`;
 // eslint-disable-next-line max-len
 
 const IDENTITY_CONFIG = {
     authority: authorityBaseUrl,
     client_id: process.env.REACT_APP_IDENTITY_CLIENT_ID,
     redirect_uri: `${appBaseUrl}/logincallback`,
     login: `${authorityUrl}`,
+    register: authorityRegisterUrl,
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/identity-config.ts:14]: This should not be here. Move it to `urls`. This file is mostly redundant and is on its way to deletion.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import { LinkButton, LinkButtonType } from '../guidelines/buttons/Button';
+import { IDENTITY_CONFIG } from '../../identity-config';
+import styles from './LoginForm.module.scss';
+
+const LoginPage = () => {
+    const { setUsername, setPassword, signIn, loginErrorMessage } = useAuth();
+    const usernameFieldName = 'Username';
+    const passwordFieldName = 'Password';
+
+    const onChangeUpdateUsername = useCallback((value: any) => {
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx:17]: `handle`

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        setUsername(value);
+    }, [ setUsername ]);
+
+    const onChangeUpdatePassword = useCallback((value: any) => {
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx:21]: `handle`

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    }, [ signIn ]);
+
+    const renderLoginErrorMessage = useCallback(() => {
+        if (isNil(loginErrorMessage)) {
+            return null;
+        }
+
+        return (<span className={styles.errorMessage}>{loginErrorMessage}</span>);
+    }, [ loginErrorMessage ]);
+    return (
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx:36]: empty line.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    const handleLoginClick = useCallback(async () => {
+        await signIn();
+    }, [ signIn ]);
+
+    const renderLoginErrorMessage = useCallback(() => {
+        if (isNil(loginErrorMessage)) {
+            return null;
+        }
+
+        return (<span className={styles.errorMessage}>{loginErrorMessage}</span>);
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx:34]: Ternary here?

**Review by Minkov — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/identity-config.ts (hunk 3 above)
Comment (shsimeonova) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/identity-config.ts:14]: I suggest i log this into an issue as there are changes needed in the index.cshtml file for this. We might log a bigger issue to remove this file and fix this in it.

Comment (Minkov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/identity-config.ts:14]: Don't remove it now. Just move the `url`

**Review by Minkov — DISMISSED** [review]

**Review by Minkov — APPROVED** [review]

**Pull request merged**